src/tribler-core/tribler_core/components/tag/rules/tag_rules.py

```diff
@@ -0,0 +1,42 @@
+import re
+from typing import AnyStr, Iterable, Optional, Pattern, Sequence
+
+from tribler_core.components.tag.community.tag_validator import is_valid_tag
+
+square_brackets = re.compile(r'\[([^\[\]]+)]')
+brackets = re.compile(r'\(([^()]+)\)')
```

Comment:
**nitpick:** should be called "parentheses" instead. Or better, something like "parentheses_re".

Reply:
Fixed.